### PR TITLE
Expose list of configured exporters 

### PR DESCRIPTION
**Pull request by LDaneliukas**
PR for #15 

image

### CODE REVIEW

**Comment by tdabasinskas** [comment]
Hi, @tcolgate ,

Were you able to look into this PR? Just curious if this project is still alive 🙂

File: main.go [hunk 1]
```diff
@@ -289,6 +290,17 @@ func (cfg *config) doProxy(w http.ResponseWriter, r *http.Request) {
 	h.ServeHTTP(w, r)
 }
 
+func (cfg *config) listModules(w http.ResponseWriter, r *http.Request) {
+	log.Debugf("Listing modules")
```
Comment (tcolgate) [inline main.go]: It might be better to use html/template here to properly escape the string content being injected.

**Comment by tcolgate** [comment]
Hi,
  very sorry I missed this. This looks like a great idea. I put a comment on the PR, basically, probably better to either escape the text with whatever html/template uses, or use html/template directly.

**Comment by LDaneliukas** [comment]
Hey, @tcolgate - glad to see this is still maintained, we're quite heavy users of it :)

Is the PR good now?

File: README.md [hunk 2]
```diff
@@ -101,3 +104,10 @@ scrape_configs:
     static_configs:
       - targets: ['host:9999']
 ```
+
+## Building / releasing
```
Comment (tcolgate) [inline README.md]: Drop these, they aren't relevant to this PR, I've already tweaked in the README a bit, I'll add these, or equivalent in there directly.

File: main.go [hunk 3]
```diff
@@ -289,6 +291,23 @@ func (cfg *config) doProxy(w http.ResponseWriter, r *http.Request) {
...
+	w.Header().Set("Content-Type", "text/html; charset=utf-8")
+	tmpl, err := template.New("modules").Parse(`
+		<h2>Exporters:</h2>
+			<ul>
+				{{range $name, $cfg := .Modules}}
+					<li><a href="/proxy?module={{$name}}">{{$name}}</a></li>
+				{{end}}
+			</ul>`)
+	if err != nil {
```
Comment (tcolgate) [inline main.go]: This should never fail, so I'm tempted to say you can just _ out the err above. Failing that, if we check the error here, then please http.Error(....) and return (you are currently exeuting the template even if this fails).

File: main.go [hunk 4]
```diff
@@ -289,6 +291,23 @@ func (cfg *config) doProxy(w http.ResponseWriter, r *http.Request) {
...
+		log.Error(err)
+	}
+	tmpl.Execute(w, cfg)
```
Comment (tcolgate) [inline main.go]: Please check the error from Execute.

**Comment by LDaneliukas** [comment]
Update the PR. There's also quite a few of go.mod changes, due to `go mod tidy`, I hope that's okey.

**Comment by tcolgate** [comment]
The PR is showing conflicts, I'm just trying to rebase locally.

**Comment by LDaneliukas** [comment]
Rebased the fork against upstream

**Comment by tcolgate** [comment]
Great, sorry that got a bit painful, the PR was trying to rip out a bunch of fixes.
Thanks, this is a really nice addition.

**Comment by candlerb** [comment]
Any chance of a new binary release?  This change was merged shortly after v0.2.9 was released.  Thanks!

**Comment by tcolgate** [comment]
I'll look into this on Monday.

**Pull request merged**